.github/workflows/check-links.yml

```
@@ -0,0 +1,49 @@
+name: Check docs links
+on:
+  pull_request:
+    branches: main
+    paths:
+      - docs/**
+
+jobs:
+  check-docs-links:
+    runs-on: ubuntu-latest
+    name: Check links
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install python
+        uses: actions/setup-python@v2
+
+      - name: Install python dependencies
+        run: |
+          pip install -r docs/requirements.txt
+
+      - name: Find all the broken links
+        id: find_links
+        run: |
+          # Build the site and save log output
+          mkdocs build &> build-log.txt
+
+          # Capture all the bad links
+          grep "relative link" build-log.txt > bad_links.txt
+
+          # Save the number of bad links to github output
+          n_bad_links=`wc -l bad_links.txt | cut -f 1 -d " "`
+          echo "n_bad_links=$n_bad_links" >> $GITHUB_OUTPUT
```

You can do this in 1: 
```
          # Save the number of bad links to github output
          echo "n_bad_links=$(grep "relative link" build-log.txt | wc -l)" >> $GITHUB_OUTPUT
```

The trick here is that `wc -l` doesn't print a file name if it is working on stdin. Also `$()` for command substitution.

But I see you want the file, so the real suggestion is:
```suggestion
          grep "relative link" build-log.txt > bad_links.txt

          # Save the number of bad links to github output
          echo "n_bad_links=$(wc -l < bad_links.txt)" >> $GITHUB_OUTPUT
```